src/WebJobs.Extensions.DurableTask/Scale/DurableTaskTriggersScaleProvider.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#nullable enable
+#if FUNCTIONS_V3_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Scale
+{
+    internal class DurableTaskTriggersScaleProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly IScaleMonitor monitor;
+        private readonly ITargetScaler targetScaler;
+
+        public DurableTaskTriggersScaleProvider(
+            IServiceProvider serviceProvider,
+            TriggerMetadata triggerMetadata)
+        {
+            string functionId = triggerMetadata.FunctionName;
+            FunctionName functionName = new FunctionName(functionId);
+
+            DurableTaskOptions options = this.GetOptions(serviceProvider, triggerMetadata);
+
+            IDurabilityProviderFactory durabilityProviderFactory = this.GetDurabilityProviderFactory(serviceProvider, options);
+            DurabilityProvider defaultDurabilityProvider = durabilityProviderFactory.GetDurabilityProvider();
+
+            string? connectionName = durabilityProviderFactory is AzureStorageDurabilityProviderFactory azureStorageDurabilityProviderFactory
+                ? azureStorageDurabilityProviderFactory.DefaultConnectionName
+                : null;
+
+            this.targetScaler = ScaleUtils.GetTargetScaler(
+                defaultDurabilityProvider,
+                functionId,
+                functionName,
+                connectionName,
+                options.HubName);
+
+            this.monitor = ScaleUtils.GetScaleMonitor(
+                defaultDurabilityProvider,
+                functionId,
+                functionName,
+                connectionName,
+                options.HubName);
+        }
+
+        private DurableTaskOptions GetOptions(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            var nameResolver = serviceProvider.GetService<INameResolver>();
```

Comment:
Minor nit: but I think it's consider best practice to have dependency-injected services like `INameResolver` and `IOptions<DurableTaskOptions>` as explicit constructor parameters instead of using `IServiceProvider`.